index.tt

```diff
@@ -6,152 +6,349 @@
-          <i class="fa fa-chevron-right"></i> Learn more about NixOS
-        </a>
+      <div class="span6">
+
+        <h3>Isolated development environments</h3>
+        <p>
+          After you get familiar with <code>nix-shell -p</code> you can take
+          the next step further and <a href="[%root%]learn.html">learn some
+            Nix</a>. To setup a more persistent environment you can also write a
+          simple <code>shell.niix</code> file:
+        </p>
+        <pre class="well" style="margin-top:1em">
+{ pkgs ? import &lt;nixpkgs&gt; {}
+}:
+pkgs.mkShell {
+  name = "dev-shell";
+  buildInputs = [
+    pkgs.python3
+    pkgs.python3Packages.virtualenv
+    pkgs.nodejs
+    pkgs.yarn
+  ];
+}</pre>
+        <p>Then enter development environment with:</p>
+        <pre class="well" style="margin-top:1em">
+<span class="nix-command"><strong>nix-shell</strong></span>
+(nix-shell)<span class="nix-command">virtualenv --version</span>
+16.7.9
+(nix-shell)<span class="nix-command">yarn --version</span>
+1.22.4</pre>
+        <p>
+          Commit the above <code>shell.nix</code> file and let you coworkers have
+          easier time setting their development environment.
+        </p>
+
       </div>
     </div>
-  </div></div>
-  
-  <hr />
-  
-  <h2>More about NixOS</h2>
-  
-  <div class="row-fluid more-about">
-    <div class="span4">
-      <h3>Nixpkgs</h3>
-      <p>
-        The Nix Packages collection (Nixpkgs) is the ever growing
-        packages collection for the Nix package manager, released
-        under a permissive MIT/X11 license.
-      </p>
-      <a class="learn-more" href="[%root%]nixpkgs">
-        More about Nixpkgs
-        <i class="fa fa-angle-right"></i>
-      </a>
-    </div>
-    <div class="span4">
-      <h3>NixOps</h3>
-      <p>
-        NixOps is a tool for deploying sets of NixOS Linux machines,
-        bare-metal or to the cloud. It extends NixOS’s declarative
-        approach to system configuration management to networks and
-        provisioning.
-      </p>
-      <a class="learn-more" href="https://github.com/NixOS/nixops">
-        More about NixOps
-        <i class="fa fa-angle-right"></i>
-      </a>
+  </section>
+  <section class="tab-pane" id="build">
+    <div class="row-fluid">
+      <div class="span6">
+        <h3>Minimal docker image</h3>
+        <p>
+          Declarative way to build minimal docker images. No build tools inside
+          docker image, no complex multi stage build process, only what your
+          application needs.
+        </p>
+        <p>
+          The following Nix expression (<code>default.nix</code>) defines a
+          docker image with <strong>only</strong> Python 3 installed in it.
+        </p>
+        <pre class="well" style="margin-top:1em">
+{ pkgs ? import &lt;nixpkgs&gt; {}
+}:
+pkgs.dockerTools.buildLayeredImage {
+  name = "only-python";
+  contents = [ pkgs.python3 ];
+}</pre>
```

Comment:
a better example would be something like a container you would actually build and deploy, like a webservice

so we could add one additional like like `config.Cmd = [ "/bin/hello" ];` and use curl in the example